app/models/user.rb

```diff
@@ -54,6 +54,17 @@ class User < ApplicationRecord
   validates :full_name, :gender, presence: true
   validates :gender, inclusion: { in: genders }
 
+  def self.from_social_provider(provider, user_params)
+    find_or_create_by!(provider: provider, uid: user_params['id']) do |user|
+      user.password = Devise.friendly_token[0, 20]
+      user.full_name = user_params['first_name'] + ' ' + user_params['last_name']
+      user.email = user_params['email']
+      # user.avatar = user_params['picture']
```

Comment:
is this commented because the PR with the avatar is still in Code Review? or why?

Reply:
Exactly, user avatar PR is still not merged.